Source/Public/Set-VSTeamWorkItemType.ps1

```diff
@@ -0,0 +1,90 @@
+function Set-VSTeamWorkItemType {
+   [CmdletBinding(DefaultParameterSetName='LeaveAlone',SupportsShouldProcess=$true,ConfirmImpact='High')]
+   param(
+      [parameter(ValueFromPipelineByPropertyName=$true)]
+      [vsteam_lib.ProcessTemplateValidateAttribute()]
+      [ArgumentCompleter([vsteam_lib.ProcessTemplateCompleter])]
+      $ProcessTemplate = $env:TEAM_PROCESS,
+
+      [parameter(Mandatory = $true,ValueFromPipelineByPropertyName=$true, Position=0)]
+      [ArgumentCompleter([vsteam_lib.WorkItemTypeCompleter])]
+      [Alias('Name')]
+      $WorkItemType,
+
+      [string] $Description,
+
+      [ArgumentCompleter([vsteam_lib.ColorCompleter])]
+      [vsteam_lib.ColorTransformToHexAttribute()]
+      $Color,
+
+      [vsteam_lib.IconTransformAttribute()]
+      [ArgumentCompleter([vsteam_lib.IconCompleter])]
+      [string]$Icon ,
+
+      [parameter(ParameterSetName='Hide', Mandatory = $true)]
+      [switch]$Disabled,
+
+      [parameter(ParameterSetName='Show', Mandatory = $true)]
+      [switch]$Enabled,
+
+      [switch]$Force
+   )
+   process {
+      #Get the existing workitem-type definition. Drop out if can't be found 
+      $wit  = Get-VSTeamWorkItemType -ProcessTemplate $ProcessTemplate -WorkItemType $WorkItemType
+      if (-not $wit) {Write-Warning "'$workitemType' does not appear to be a valid Workitem type." ; return}
+      else           {$url  = $wit.url + "?api-version=" + (_getApiVersion Processes)  }
+
+#region set up the contents to go in the JSON body
+      $body = @{} #note. "Changing" some options to their current value has caused errors but those below seem OK...
```

Comment:
is it needed that you have to post all the existing values? In some cases, the API is fine just posting what has changed.

Reply:
I have had some cases where re-posting caused problems. In other cases things fail if the name is not included in the body it's quite some time since I wrote that piece and I can't be sure if was needed or if it was a habit from where I always had `write-debug $The_JSON_body` and preferred to have all the fields visible.